src/components/Synthetics/TableMarketFilter/MarketFilterBase.tsx

```diff
@@ -0,0 +1,99 @@
+import { t } from "@lingui/macro";
+import { ReactNode, useCallback, useMemo } from "react";
+
+import { useMarketsInfoData } from "context/SyntheticsStateContext/hooks/globalsHooks";
+
+import { getNormalizedTokenSymbol } from "config/tokens";
+import { useMarketTokensData } from "domain/synthetics/markets/useMarketTokensData";
+import { getMarketIndexName, getMarketPoolName } from "domain/synthetics/markets/utils";
+import useSortedPoolsWithIndexToken from "domain/synthetics/trade/useSortedPoolsWithIndexToken";
+import { useChainId } from "lib/chains";
+
+import { TableOptionsFilter } from "components/Synthetics/TableOptionsFilter/TableOptionsFilter";
+import type { Item } from "components/Synthetics/TableOptionsFilter/types";
+import TokenIcon from "components/TokenIcon/TokenIcon";
+
+export type MarketFilterBaseProps = {
+  /**
+   * Market addresses
+   */
+  value: string[];
+  /**
+   * Callback when the market selection changes
+   */
+  onChange: (value: string[]) => void;
+  excludeSpotOnly?: boolean;
+  beforeContent?: ReactNode | undefined;
+  extraIsActive?: boolean;
+};
+
+export function MarketFilterBase({
+  value,
+  onChange,
+  excludeSpotOnly,
+  beforeContent,
+  extraIsActive,
+}: MarketFilterBaseProps) {
+  const marketsInfoData = useMarketsInfoData();
+  const { chainId } = useChainId();
+  const { marketTokensData: depositMarketTokensData } = useMarketTokensData(chainId, { isDeposit: true });
+  const { marketsInfo: markets } = useSortedPoolsWithIndexToken(marketsInfoData, depositMarketTokensData);
+
+  const marketsOptions = useMemo<Item<string>[]>(() => {
+    return markets
+      .filter((market) => {
+        if (excludeSpotOnly !== undefined) {
+          return !market.isSpotOnly;
+        }
+
+        return true;
+      })
+      .map((market) => {
+        return {
+          text: market.name,
+          data: market.marketTokenAddress,
+        };
+      });
+  }, [excludeSpotOnly, markets]);
+
+  const ItemComponent = useCallback(
```

Comment:
why not just make it a regular functional component?

Reply:
an altrnative would be 
<img width="861" alt="image" src="https://github.com/gmx-io/gmx-interface/assets/155806380/d5206a5b-cb5f-42be-a5bc-6ca4e6bf1c49">

i dont think its necessary. isnt it good enough ?